test/framework/clusterctl/repository.go

```diff
@@ -168,3 +168,33 @@ func YAMLForComponentSource(ctx context.Context, source ProviderVersionSource) (
 
 	return data, nil
 }
+
+// getComponentSourceFromURL fetches contents of component source YAML file from provided URL source.
+func getComponentSourceFromURL(source ProviderVersionSource) ([]byte, error) {
+	var buf []byte
+
+	u, err := url.Parse(source.Value)
+	if err != nil {
+		return nil, err
+	}
+
+	switch u.Scheme {
+	case fileURIScheme:
+		buf, err = ioutil.ReadFile(u.Path)
```

Comment:
In the test framework we are making all the relative path are processed starting from the config file path 

https://github.com/kubernetes-sigs/cluster-api/blob/57c6d549ec1ffabf93b460e51efdeca856d97c1f/test/framework/clusterctl/e2e_config.go#L281

Should we do the same for relative urls of type file?

Reply:
I think that would be very convenient. But in this case, i suppose, documentation also needs to be updated, to notify user of that behavior. Also, what the syntax would for relative paths? Does `file://./path/ti/file` looks ok?

Reply:
now that file: is optional I think this makes even more sense.
wrt to the syntax, we should rely on conventions already implemented in filepath.IsAbs, but given if this turns out to be complicated, fell free to drop the idea given this is a nit and there are already other ways to use relative paths

Reply:
Added support for relative paths.